npm/private/transitive_closure.bzl

```diff
@@ -40,7 +40,12 @@ def gather_transitive_closure(packages, no_optional, direct_deps, transitive_clo
             if version in transitive_closure[name]:
                 continue
             transitive_closure[name].insert(0, version)
-            package_info = packages[package_key]
+            if package_key.startswith("link:"):
+                # we don't need to drill down through first-party links for the transitive closure since there are no cycles
+                # allowed in first-party links
+                continue
```

Comment:
nit: `continue` statements can be hard to follow, looks like you could just invert the condition and avoid that?